listenbrainz_spark/recommendations/recommend.py

```diff
@@ -33,6 +41,149 @@ def load_model(path):
     """
     return MatrixFactorizationModel.load(listenbrainz_spark.context, path)
 
+def convert_recommendation_to_dict_for_html(row):
+    """ Convert recommendation to dictionary for HTML file.
+
+        Args:
+            row : PySpark row object.
+
+        Returns:
+            Dictionary of a recommendation.
+    """
+    return {
+                'artist_name': row.artist_name,
+                'mb_artist_credit_id': row.mb_artist_credit_id,
+                'mb_artist_credit_mbids': row.mb_artist_credit_mbids,
+                'mb_recording_mbid': row.mb_recording_mbid,
+                'mb_release_mbid': row.mb_release_mbid,
+                'release_name': row.release_name,
+                'track_name': row.track_name,
+    }
+
+def add_user_name_to_recommendation(recommendation, user_name):
+    """ Include user name in recommendation.
+
+        Args:
+            recommendation (dict): Recommendation for a user.
+            user_name (str): User name of the user.
+
+        Returns:
+            Dictionary containing recommendation and user name.
+    """
+    return {
+                'artist_name': recommendation['artist_name'],
+                'mb_artist_credit_id': recommendation['mb_artist_credit_id'],
+                'mb_artist_credit_mbids': recommendation['mb_artist_credit_mbids'],
+                'mb_recording_mbid': recommendation['mb_recording_mbid'],
+                'mb_release_mbid': recommendation['mb_release_mbid'],
+                'release_name': recommendation['release_name'],
+                'track_name': recommendation['track_name'],
+                'user_name': user_name,
+    }
+
+def get_archive_name(to_date, from_date, current_date):
+    """ Get name of the archive to save candidate recordings.
+
+        Args:
+            to_date (datetime): Date from which listens were fetched to generate candidate sets.
+            from_date (datetime): Date upto which listens were fetched to generate candidate sets.
+            current_date (datetime): Current date.
+
+        Returns:
+            archive_name (str): Candidate recordings archive name.
+    """
+    archive_name = 'listenbrainz-candidate-recordings-dump-{from_date}-{to_date}-{curr_date}'.format(
+                        from_date=from_date.strftime('%Y%m%d'), to_date=to_date.strftime('%Y%m%d'),
+                        curr_date=current_date.strftime('%Y%m%d-%H%M%S')
+                    )
+    return archive_name
+
+def dump_candidate_recordings(recommendations, temp_file, playlist_type):
+    """ Write candidate recordings to a file based on type of playlist.
+
+        Args:
+            recommendations (dict): Recommended recordings for users.
+            temp_file (str) : File path to dump recordings.
+            playlist_type (str): Type of playlist.
+    """
+
+    with open(temp_file, 'w') as f:
```

Comment:
One thing that I really like doing when generating dumps is to have python compress the data as it is being output. It makes the process a bit slower, but overall it saves time from having to compress the output in a second step. See https://docs.python.org/3/library/lzma.html

Reply:
Though my point may be moot, since you do use the very intelligent multithreaded pxz compression below. Perhaps this comment really isn't actionable. 😇